redis/src/cluster_async/mod.rs

```diff
@@ -94,70 +94,96 @@ where
 type ConnectionFuture<C> = future::Shared<BoxFuture<'static, C>>;
 type ConnectionMap<C> = HashMap<String, ConnectionFuture<C>>;
 
-struct ClusterConnInner<C> {
-    connections: ConnectionMap<C>,
+struct ConnectionsAndMetadata<C> {
```

Comment:
With apologies for the inconvenience here, I feel like adding this struct and related connection-handling refactorings are in themselves a large change-set. Is it possible to bring those in first in a separate PR? :pray:

Reply:
https://github.com/redis-rs/redis-rs/pull/864

Reply:
also https://github.com/redis-rs/redis-rs/pull/842